agent/src/main/java/com/teamscale/jacoco/agent/testimpact/TestwiseCoverageAgent.java

```diff
@@ -86,7 +90,10 @@ private String handleTestEnd(Request request, Response response) throws DumpExce
 
 		if (!request.body().isEmpty()) {
 			try {
-				TestExecution testExecution = GSON.fromJson(request.body(), TestExecution.class);
+				TestExecution testExecution = testExecutionJsonAdapter.fromJson(request.body());
+				if (testExecution == null) {
```

Comment:
Please comment in which case this might happen and why the empty string is the appropriate return value in that case.